homeassistant/components/media_player/mediaroom.py

```diff
@@ -9,134 +9,170 @@
 import voluptuous as vol
 
 from homeassistant.components.media_player import (
-    MEDIA_TYPE_CHANNEL, SUPPORT_PAUSE, SUPPORT_PLAY_MEDIA,
-    SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_STOP, PLATFORM_SCHEMA,
-    SUPPORT_NEXT_TRACK, SUPPORT_PREVIOUS_TRACK, SUPPORT_PLAY,
-    SUPPORT_VOLUME_STEP, SUPPORT_VOLUME_MUTE,
-    MediaPlayerDevice)
+    MEDIA_TYPE_CHANNEL, SUPPORT_PAUSE, SUPPORT_PLAY_MEDIA, SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON, SUPPORT_STOP, PLATFORM_SCHEMA, SUPPORT_NEXT_TRACK,
+    SUPPORT_PREVIOUS_TRACK, SUPPORT_PLAY, SUPPORT_VOLUME_STEP,
+    SUPPORT_VOLUME_MUTE, MediaPlayerDevice,
+)
+from homeassistant.helpers.dispatcher import (
+    dispatcher_send, async_dispatcher_connect
+)
 from homeassistant.const import (
-    CONF_HOST, CONF_NAME, CONF_OPTIMISTIC, CONF_TIMEOUT,
-    STATE_PAUSED, STATE_PLAYING, STATE_STANDBY,
-    STATE_ON)
+    CONF_HOST, CONF_NAME, CONF_OPTIMISTIC, STATE_OFF,
+    CONF_TIMEOUT, STATE_PAUSED, STATE_PLAYING, STATE_STANDBY,
+    STATE_UNAVAILABLE
+)
 import homeassistant.helpers.config_validation as cv
-REQUIREMENTS = ['pymediaroom==0.5']
+
+REQUIREMENTS = ['pymediaroom==0.6']
 
 _LOGGER = logging.getLogger(__name__)
 
-NOTIFICATION_TITLE = 'Mediaroom Media Player Setup'
-NOTIFICATION_ID = 'mediaroom_notification'
+PLATFORM = "mediaroom"
 DEFAULT_NAME = 'Mediaroom STB'
 DEFAULT_TIMEOUT = 9
 DATA_MEDIAROOM = "mediaroom_known_stb"
-
-SUPPORT_MEDIAROOM = SUPPORT_PAUSE | SUPPORT_TURN_ON | SUPPORT_TURN_OFF | \
-    SUPPORT_VOLUME_STEP | SUPPORT_VOLUME_MUTE | \
-    SUPPORT_PLAY_MEDIA | SUPPORT_STOP | SUPPORT_NEXT_TRACK | \
-    SUPPORT_PREVIOUS_TRACK | SUPPORT_PLAY
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_HOST): cv.string,
-    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-    vol.Optional(CONF_OPTIMISTIC, default=False): cv.boolean,
-    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
-})
-
-
-def setup_platform(hass, config, add_devices, discovery_info=None):
+SIGNAL_STB_NOTIFY = 'stb_discovered'
+SUPPORT_MEDIAROOM = SUPPORT_PAUSE | SUPPORT_TURN_ON | SUPPORT_TURN_OFF \
+    | SUPPORT_VOLUME_STEP | SUPPORT_VOLUME_MUTE | SUPPORT_PLAY_MEDIA \
+    | SUPPORT_STOP | SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK \
+    | SUPPORT_PLAY
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
Please try not to do style / code changes as it dirties the diff. This here is not changed, but it is all marked as changed because you changed indentation.